src/query/block/container_test.go

```diff
@@ -0,0 +1,403 @@
+
+	ex := 1.0
+	for it.Next() {
+		current := it.Current()
+		assert.Equal(t, numSteps, current.Len())
+		for _, ac := range current.Values() {
+			assert.Equal(t, ex, ac)
+		}
+
+		assert.Equal(t, []byte(fmt.Sprint(ex)), current.Meta.Name)
+		ex++
+	}
+
+	metas := it.SeriesMeta()
+	assert.Equal(t, 2, len(metas))
+	assert.Equal(t, []byte("1"), metas[0].Name)
+	assert.Equal(t, []byte("2"), metas[1].Name)
+
+	assert.NoError(t, it.Err())
+	assert.NotPanics(t, func() { it.Close() })
+}
+
+func buildUnconsolidatedStepBlock(ctrl *gomock.Controller,
+	v float64, first bool) Block {
+	base := NewMockBlock(ctrl)
+	b := NewMockUnconsolidatedBlock(ctrl)
+	base.EXPECT().Unconsolidated().Return(b, nil)
+
+	meta := Metadata{
+		Tags:   models.NewTags(0, models.NewTagOptions()),
+		Bounds: containerBounds,
+	}
+
+	base.EXPECT().Meta().Return(meta).AnyTimes()
+	it := NewMockUnconsolidatedStepIter(ctrl)
+	it.EXPECT().Close()
+	it.EXPECT().Err().Return(nil).AnyTimes()
+	it.EXPECT().Next().Return(true).Times(numSteps)
+	it.EXPECT().Next().Return(false)
+	for i := 0; i < numSteps; i++ {
+		s := NewMockUnconsolidatedStep(ctrl)
+		tt := now.Add(time.Duration(i) * step)
+		if first {
+			s.EXPECT().Time().Return(tt)
+		}
+
+		s.EXPECT().Values().Return([]ts.Datapoints{
+			ts.Datapoints{
+				ts.Datapoint{Timestamp: tt, Value: v},
+				ts.Datapoint{Timestamp: tt, Value: v},
+			},
+		})
+
+		it.EXPECT().Current().Return(s)
+	}
+
+	it.EXPECT().SeriesMeta().
+		Return([]SeriesMeta{{Name: []byte(fmt.Sprint(v))}}).Times(2)
+	if first {
+		it.EXPECT().StepCount().Return(numSteps)
+	}
+
+	b.EXPECT().StepIter().Return(it, nil)
+	return base
+}
+
+func TestUnconsolidatedContainerStepIter(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	block := buildUnconsolidatedStepBlock(ctrl, 1, true)
+	blockTwo := buildUnconsolidatedStepBlock(ctrl, 2, false)
+
+	c, err := NewContainerBlock(block, blockTwo)
+	require.NoError(t, err)
+
+	consolidated, err := c.Unconsolidated()
+	require.NoError(t, err)
+
+	// FIXME to use better equality arnikola
+	assert.Equal(t, containerBounds, consolidated.Meta().Bounds)
+	assert.Equal(t, opts, consolidated.Meta().Tags.Opts)
+
+	it, err := consolidated.StepIter()
+	require.NoError(t, err)
+
+	count := 0
+	for it.Next() {
+		st := it.Current()
+		tt := now.Add(step * time.Duration(count))
+		ex := []ts.Datapoints{
+			ts.Datapoints{
+				ts.Datapoint{Timestamp: tt, Value: 1},
+				ts.Datapoint{Timestamp: tt, Value: 1},
+			},
+			ts.Datapoints{
+				ts.Datapoint{Timestamp: tt, Value: 2},
+				ts.Datapoint{Timestamp: tt, Value: 2},
+			},
+		}
+
+		assert.Equal(t, ex, st.Values())
+		count++
+	}
+
+	assert.Equal(t, count, numSteps)
+
+	metas := it.SeriesMeta()
+	assert.Equal(t, 2, len(metas))
+	assert.Equal(t, []byte("1"), metas[0].Name)
+	assert.Equal(t, []byte("2"), metas[1].Name)
+	assert.Equal(t, numSteps, it.StepCount())
+
+	assert.NoError(t, it.Err())
+	assert.NotPanics(t, func() { it.Close() })
+}
+
+//
+//
+//
+//
```

Comment:
Probably can remove these "//" before we merge?

Reply:
RIP LOC stats 😭